servers/rendering/renderer_rd/shader_rd.cpp

```diff
@@ -513,9 +515,17 @@ void ShaderRD::_compile_version(Version *p_version, int p_group) {
 	compile_data.group = p_group;
 
 #if 1
-	WorkerThreadPool::GroupID group_task = WorkerThreadPool::get_singleton()->add_template_group_task(this, &ShaderRD::_compile_variant, &compile_data, group_to_variant_map[p_group].size(), -1, true, SNAME("ShaderCompilation"));
-	WorkerThreadPool::get_singleton()->wait_for_group_task_completion(group_task);
-
+	if (group_to_variant_map[p_group].size() > 0) {
+		bool first_result = _compile_variant(0, &compile_data);
```

Comment:
> I think instead of using `0` you can use `group_to_variant_map[p_group].size() - 1`
Great idea! 

> that this will still have a performance impact as you are limiting the amount of parallelism possible

Maybe I can make this change a editor only feature, that way it won't affect the performance of runtime.